hyperframe/frame.py

```diff
@@ -110,7 +113,15 @@ def parse_frame_header(header):
         stream_id = fields[4]
 
         if type not in FRAMES:
-            raise UnknownFrameError(type, length)
+            if strict:
+                raise UnknownFrameError(type, length)
+
+            frame = ExtensionFrame(
+                type=type,
+                flag_byte=flags,
+                stream_id=stream_id,
+            )
+            return (frame, length)
```

Comment:
So, I don't think we can actually use `parse_flags` here. `parse_flags` depends on us having a definition of which flags are available. Since the meaning of the flags is defined on a per-frame basis, the base `ExtensionFrame` doesn't have the ability to do that.

The `assign_flag_mapping` method lets us do that after the fact - an upstream consumer, which is interpreting the frame, can insert a mapping between bit location and flag name, resulting in the frame getting a `flags` member.